rbac/management/group/view.py

```diff
@@ -510,13 +522,22 @@ def principals(self, request, uuid=None):
             serializer = GroupPrincipalInputSerializer(data=request.data)
             if serializer.is_valid(raise_exception=True):
                 principals = serializer.data.pop("principals")
+
+            # Filter the service accounts from the principals so that we only
+            # attempt to save the latter.
+            serviceAccounts = self.__filterServiceAccountsFromPrincipals(principals)
+
             if settings.AUTHENTICATE_WITH_ORG_ID:
                 resp = self.add_principals(group, principals, org_id=org_id)
             else:
                 resp = self.add_principals(group, principals, account=account)
             if isinstance(resp, dict) and "errors" in resp:
                 return Response(status=resp["status_code"], data=resp["errors"])
             output = GroupSerializer(resp)
+
+            # Append the service accounts back to the payload.
+            output.appendServiceAccounts(serviceAccounts)
+
```

Comment:
Is this able to go away from this PR now that we're mocking based on the query param alone on the `GET`?

Reply:
Removed!